awx/playbooks/project_update.yml

```diff
@@ -23,88 +25,111 @@
         state: absent
       when: scm_delete_on_update|default('')
 
-    - name: update project using git and accept hostkey
-      git:
-        dest: "{{project_path|quote}}"
-        repo: "{{scm_url}}"
-        version: "{{scm_branch|quote}}"
-        force: "{{scm_clean}}"
-        accept_hostkey: "{{scm_accept_hostkey}}"
-        #clone: "{{ scm_full_checkout }}"
-        #update: "{{ scm_full_checkout }}"
-      when: scm_type == 'git' and scm_accept_hostkey is defined
-      register: scm_result
-
-    - name: Set the git repository version
-      set_fact:
-        scm_version: "{{ scm_result['after'] }}"
-      when: "'after' in scm_result"
-
-    - name: update project using git
-      git:
-        dest: "{{project_path|quote}}"
-        repo: "{{scm_url}}"
-        version: "{{scm_branch|quote}}"
-        force: "{{scm_clean}}"
-        #clone: "{{ scm_full_checkout }}"
-        #update: "{{ scm_full_checkout }}"
-      when: scm_type == 'git' and scm_accept_hostkey is not defined
-      register: scm_result
-
-    - name: Set the git repository version
-      set_fact:
-        scm_version: "{{ scm_result['after'] }}"
-      when: "'after' in scm_result"
-
-    - name: update project using hg
-      hg:
-        dest: "{{project_path|quote}}"
-        repo: "{{scm_url|quote}}"
-        revision: "{{scm_branch|quote}}"
-        force: "{{scm_clean}}"
-        #clone: "{{ scm_full_checkout }}"
-        #update: "{{ scm_full_checkout }}"
+    - block:
+      - name: check repo using git
+        git:
+          dest: "{{project_path|quote}}"
+          repo: "{{scm_url}}"
+          version: "{{scm_branch|quote}}"
+          force: "{{scm_clean}}"
+          update: false
+          clone: false
+        register: repo_check
+        when: scm_full_checkout|default('')
+        ignore_errors: true
+
+      - name: break if already checked out
+        meta: end_play
+        when: scm_full_checkout|default('') and repo_check|succeeded and repo_check.before == scm_branch
```

Comment:
I'm wondering if this `before == scm_branch` is going to work.

`before` and `after` are hashes and sometimes `scm_branch` is going to be a tag/branch... is this just a shortcut for if we have specified a particular revision?

Reply:
AIUI, `scm_branch` for check jobs is a tag/branch, `scm_branch` for run jobs is the revision.

Reply:
ah it is indeed